lambda.py

```diff
@@ -70,6 +76,63 @@ def xmlverifier(event, context):
             event['local'], event['remote']))
 
 
+DEFAULT_COLLECTIONS = [
+    {'bucket': 'blocklists',
+     'collection': 'certificates'},
+    {'bucket': 'blocklists',
+     'collection': 'addons'},
+    {'bucket': 'blocklists',
+     'collection': 'plugins'},
+    {'bucket': 'blocklists',
+     'collection': 'gfx'},
+    {'bucket': 'pinning',
+     'collection': 'pins'}
+]
+
+
+def validate_signature(event, context):
+    server_url = event['server']
+    collections = event.get('collections', DEFAULT_COLLECTIONS)
+
+    for collection in collections:
+        client = Client(server_url=server_url,
+                        bucket=collection['bucket'],
+                        collection=collection['collection'])
+        print('Looking at %s: ' % client.get_endpoint('collection'), end='')
+
+        # 1. Grab collection information
+        dest_col = client.get_collection()
+
+        # 2. Grab records
+        records = client.get_records(_sort='-last_modified')
+        timestamp = client.get_records_timestamp()
+
+        # 3. Serialize
+        serialized = canonical_json(records, timestamp)
+
+        # 4. Compute hash
+        computed_hash = compute_hash(serialized)
+
+        # 5. Grab the signature
+        signature = dest_col['data']['signature']
+
+        # 6. Grab the public key
+        f = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
Shouldn't the `try-finally` include this too, since this is the first place where the file gets created?